uat/custom-components/src/test/java/com/aws/greengrass/LogGeneratorTest.java

```diff
@@ -0,0 +1,63 @@
+package com.aws.greengrass;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith({MockitoExtension.class})
+public class LogGeneratorTest {
+    @TempDir
+    static Path tempPath;
+    String logFileName = "localtest";
+    String logFileExtention = "log";
+    String logWriteFreqSeconds = "0.1";
+    String totalLogNumbers = "50";
+    String fileSizeBytes = "1024";
+    String fileSizeUnit = "KB";
+    String componentName = "com.aws.greengrass.artifacts.logGenerator";
+    String activeFileName = logFileName + "." + logFileExtention;
+
+    @Test
+    void GIVEN_request_THEN_log_file_Created()
+            throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+
+        String[] args = {logFileName, logFileExtention, fileSizeBytes, fileSizeUnit, logWriteFreqSeconds,
```

Comment:
nit: I would inline the variables here but it is ok to leave them as the class variables for now